### DISCUSSION TITLE
Pseudo USB passthrough, looking for examples or tips

### DISCUSSION THREAD

**Discussion by wiredopposite** [hathach/tinyusb#2879] [Q&A]
I've been working on class drivers that will mount a device with TUH, parse descriptors, initialize TUD, then pass the descriptors and endpoint transfers through. I have it working with some devices, but when it comes to devices that require control transfers, like HID or proprietary, I'm running into some issues. 

I'm wondering if anyone has tried to write a USB passthrough program with TinyUSB or are aware of any projects that do this, so I can get a better idea of how to handle transfers properly for this purpose.

**Comment by Mohanbandaru16** [comment]
can you able to communicate through end points ?
if so any examples ???im stuck at communication through end points.
